R/indicator_native_range_year.R

```diff
@@ -0,0 +1,117 @@
+#' Create interactive plot for counts per native region and year of introduction
+#' 
+#' Based on \code{\link{countYearProvince}} plot from grofwild
+#' @param type character, native_range level of interest should be one of 
+#' \code{c("native_continent", "native_range")} 
+#' @param xlab character string, label of the x-axis. Defaults to "year".
+#' @param ylab character string, label of the y-axis. Defaults to "number of 
+#' alien species".  
+#' @inheritParams countYearProvince
+#' @return list with:
+#' \itemize{
+#' \item{'plot': }{plotly object, for a given specie the observed number 
+#' per year and per province is plotted in a stacked bar chart}
+#' \item{'data': }{data displayed in the plot, as data.frame with:
+#' \itemize{
+#' \item{'year': }{year at which the animal was introduced}
+#' \item{'nativeRange': }{native range of the introduced animal}
+#' \item{'aantal': }{counts of animals}
+#' }
+#' }
+#' }
+#' @import plotly
+#' @importFrom reshape2 melt
+#' @importFrom INBOtheme inbo.2015.colours
+#' @export
+countYearNativerange <- function(data, jaartallen = NULL, 
+                                 type = c("native_continent", "native_range"),
+                                 width = NULL, height = NULL, 
+                                 x_lab = "year",
+                                 y_lab = "alien species",
+                                 relative = FALSE) {
+  
+  require(plotly)
+  require(data.table)
+  
+  type <- match.arg(type)
+  
+  if (is.null(jaartallen))
+    jaartallen <- sort(unique(data$first_observed))
+  
+  plotData <- data %>% 
+    mutate(native_continent = case_when(grepl(pattern = "Africa", native_range, ignore.case = TRUE) ~ "Africa",
+                                        grepl(pattern = "America", native_range, ignore.case = TRUE) ~ "America",
+                                        grepl(pattern = "Asia", native_range, ignore.case = TRUE) ~ "Asia",
+                                        grepl(pattern = "Australia", native_range, ignore.case = TRUE) ~ "Oceania",
+                                        grepl(pattern = "nesia", native_range, ignore.case = TRUE) ~ "Oceania",
+                                        grepl(pattern = "Europe", native_range, ignore.case = TRUE) ~ "Europe",
+                                        TRUE ~ as.character(NA)))
```

Comment:
agree, but as stated in https://github.com/trias-project/trias/issues/64#issuecomment-654980803 i could not/cannot find the correct column. It is not native_range because this contains lower level origins (regions).

Reply:
I understand what we should do here. We should add the mapping of this new column in the pipeline: https://github.com/trias-project/indicators/blob/master/src/01_get_data_input_checklist_indicators.Rmd where the input file for checklist indicators, [data_input_checklist_indicators.tsv](https://github.com/trias-project/indicators/blob/master/data/interim/data_input_checklist_indicators.tsv), is created based on GBIF data (taxonomc and extensions). Is it ok for you? I will do it next week.

Reply:
Ok for me ;-)

Reply:
I'll remove this part of the function. But I'll leave the conversation open until the mapping was added.

Reply:
Work on indicators is done (https://github.com/trias-project/indicators/pull/91/commits/51a06686bbd8873e35b64eb55d00db9fe2359389) and pushed on master, and doing so it is also visible on [TrIAS indicators website](https://trias-project.github.io/indicators/01_get_data_input_checklist_indicators.html#232_Add_native_content_based_on_native_range)).